File: C#-NET-Stuff/JavaExamples/SampleJavaApp-old/src/model/Account.java
package model;

import com.sun.jdi.InternalException;
import exception.UnableToAssignAccountNumberException;

import java.util.*;

public abstract class Account implements BaseAccount {
    private static Set<Integer> existingAccts;
    private static int          maxAcctNumAttemps = 10000;

    private int    acctNumber;
    private String acctOwner;
    private double balance;



    static {
        existingAccts = new TreeSet<Integer>();
    }


    public Account(String acctOwner,  double balance) throws UnableToAssignAccountNumberException {
        this.acctNumber = getUniqueAcctNumber();
        this.acctOwner = acctOwner;
        this.balance = balance;
    }

    private int getUniqueAcctNumber() throws UnableToAssignAccountNumberException {
        int newAcctNum = 0;
        Random randomizer = new Random();
        boolean foundUniqueAcctNum = false;
        int loopCount = 0;
        while ((!foundUniqueAcctNum) && (loopCount < maxAcctNumAttemps)) {
            newAcctNum = randomizer.nextInt(1000000, 999999999);
            foundUniqueAcctNum = existingAccts.add(newAcctNum);
            loopCount++;
        }
        if (foundUniqueAcctNum)
            return newAcctNum;
        else
            throw new UnableToAssignAccountNumberException("Unable to find unique account number - Account not created");
    }

    @Override
    public String toString() {
        return "Account{" +
                "acctNumber=" + acctNumber +
                ", acctOwner='" + acctOwner + '\'' +
                ", balance=" + balance +
                '}';
    }

    @Override
    public double getBalance() {
        return balance;
    };

    public int getAcctNumber() {
        return acctNumber;
    }

    public void setAcctNumber(int acctNumber) {
        this.acctNumber = acctNumber;
    }

    public String getAcctOwner() {
        return acctOwner;
    }

    public void setAcctOwner(String acctOwner) {
        this.acctOwner = acctOwner;
    }

    public void setBalance(double balance) {
        this.balance = balance;
    }

    @Override
    public abstract double withdraw(double amount);

    @Override
    public abstract double deposit(double amount);
}

File: C#-NET-Stuff/example-code/SampleCSHarp-VS2022-Net6/SampleCSharpAppTests/Model/Account.cs
using SampleCSharpAppTests.Exception;

namespace SampleCSharpAppTests.Model
{
    public abstract class Account : IBaseAccount
    {
        public const int StartingAccountNumber = 10000;
        public const int LargestAccountNumber = 30000;

        private static SortedSet<int> _existingAccts;
        private static int _maxAcctNumAttempts = LargestAccountNumber;

        private int _acctNumber;
        public string AcctOwner { get; set; }
        protected double Balance { get; set; }

        static Account()
        {
            _existingAccts = new SortedSet<int>();
        }
        public Account(string acctOwner, double balance)
        {
            this._acctNumber = GetUniqueAcctNumber();
            this.AcctOwner = acctOwner;
            this.Balance = balance;
        }

        public int GetUniqueAcctNumber()
        {
            int newAcctNum = 0;
            Random randomizer = new Random();
            bool foundUniqueAcctNum = false;
            int loopCount = 0;
            while (!foundUniqueAcctNum && loopCount < _maxAcctNumAttempts)
            {
                newAcctNum = randomizer.Next(StartingAccountNumber, LargestAccountNumber);
                foundUniqueAcctNum = _existingAccts.Add(newAcctNum);
                loopCount++;
            }
            if (foundUniqueAcctNum)
                return newAcctNum;
            else
                throw new UnableToAssignAccountNumberException("Unable to find unique account number - Account not created");
        }

        public override string ToString()
        {
            return $"Account{{ acctNumber={_acctNumber}, acctOwner='{AcctOwner}', balance={Balance} }}";
        }

        public double GetBalance()
        {
            return Balance;
        }
        public void SetBalance(double balance)
        {
            this.Balance = balance;
        }

        public int GetAcctNumber()
        {
            return _acctNumber;
        }
        public abstract double Withdraw(double amount);
        public abstract double Deposit(double amount);
    }
}
